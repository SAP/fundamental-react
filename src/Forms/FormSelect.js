import classnames from 'classnames';
import { FORM_STATES } from '../utils/constants';
import PropTypes from 'prop-types';
import React, { useEffect } from 'react';

const FormSelect = React.forwardRef(({ disabled, children, className, compact, disableStyles, state, ...props }, ref) => {

    useEffect(() => {
        if (!disableStyles) {
            require('fundamental-styles/dist/form-select.css');
        }
    }, []);

    const formSelectClasses = classnames(
        'fd-form-select',
        { [`is-${state}`]: state,
            'fd-form-select--compact': compact },
        className
    );

    return (
        <select
            {...props}
            className={formSelectClasses}
            disabled={disabled}
            ref={ref}>
            {children}
        </select>
    );
});

FormSelect.displayName = 'FormSelect';

FormSelect.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    compact: PropTypes.bool,
    disabled: PropTypes.bool,
    disableStyles: PropTypes.bool,
    state: PropTypes.oneOf(FORM_STATES)
};

export default FormSelect;
