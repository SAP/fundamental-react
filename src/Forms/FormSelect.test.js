import FormSelect from './FormSelect';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<FormSelect />', () => {
    const formSelect = (
        <FormSelect className='blue' disabled
            id='select-1'>
            <option value='1'>Duis malesuada odio volutpat elementum</option>
        </FormSelect>
    );

    test('create form item', () => {
        // create form set with form inputs
        let component = renderer.create(formSelect);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the FormSelect component', () => {
            const element = mount(<FormSelect data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });

    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <FormSelect ref={ref} />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('SELECT');
        expect(ref.current.className).toEqual('fd-form-select');
    });
});
