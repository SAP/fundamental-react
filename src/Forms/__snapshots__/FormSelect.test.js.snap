// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSelect /> create form item 1`] = `
<select
  className="fd-form-select blue"
  disabled={true}
  id="select-1"
>
  <option
    value="1"
  >
    Duis malesuada odio volutpat elementum
  </option>
</select>
`;
